lstests/core/logger.py

```diff
@@ -0,0 +1,345 @@
+#! /usr/bin/env python3
```

Comment:
Thanks for the comment. Before I started to write the code, I had done some research on the standard `logging` module, and another well-known third-party open-source library called `loguru`. While they provide more advanced features than is needed for logging the launcher service, one specific feature that is desired is however missing: able to coordinate the logging information that is sent simultaneously from multiple threads -- by _coordinate_ I mean not just printing out the lines by the chronological order, but organizing them in a human-readable manner, according to the hints given by logger's caller. Multi-threading is needed when pushing multiple actions in parallel, and when verifying transactions asynchronously. The coordinating feature is illustrated by the `_main` function of `logger.py`, and is also explained in the README doc, with an excerpt as follows:

> [With the coordinating feature enabled, the logger] will guarantee that the log file will either look like

```
thread-1: first half
thread-1: second half
thread-2: first half
thread-2: second half
```

or

```
thread-2: first half
thread-2: second half
thread-1: first half
thread-1: second half
```

but will never be of the following form

```
thread-2: first half
thread-1: first half
thread-1: second half
thread-2: second half
```

Reply:
This can be accomplished with the standard library logging module by creating a custom handler derived from `logging.handlers.BufferingHandler`.  Your derived class will be built after the pattern of the `logging.handlers.MemoryHandler` which is constructed with a `target` parameter that is the handler to which buffered, and in your case sequenced, messages will be delivered.

Reply:
While using the standard Python logging modules is preferred, for the sake of not creating too much more work to get this launcher service in, it is okay to stick with the existing logging solution implemented here.